rackcity/utils/change_planner_utils.py

```diff
@@ -178,3 +187,133 @@ def get_page_count_response_for_cp(request, change_plan):
+    return changes
+
+
+def get_cp_modification_conflicts(asset_cp):
+    conflicts = []
+    nonresolvable_message = "This conflict cannot be resolved; the change " + \
+        "needs to be removed from your change plan."
+    conflicting_asset_message_1 = "Due to more recent live changes, " + \
+        "your change plan version of this asset's "
+    conflicting_asset_message_2 = " now conflicts with a live asset. "
+    if asset_cp.asset_conflict_hostname:
+        conflicts.append({
+            "conflict_message":
+                conflicting_asset_message_1 + "hostname" +
+                conflicting_asset_message_2 + nonresolvable_message,
+            "conflicting_asset": asset_cp.asset_conflict_hostname.id,
+            "conflict_resolvable": False,
+        })
+    if asset_cp.asset_conflict_asset_number:
+        conflicts.append({
+            "conflict_message":
+                conflicting_asset_message_1 + "asset number" +
+                conflicting_asset_message_2 + nonresolvable_message,
+            "conflicting_asset": asset_cp.asset_conflict_asset_number.id,
+            "conflict_resolvable": False,
+        })
+    if asset_cp.asset_conflict_location:
+        conflicts.append({
+            "conflict_message":
+                conflicting_asset_message_1 + "rack location" +
+                conflicting_asset_message_2 + nonresolvable_message,
+            "conflicting_asset": asset_cp.asset_conflict_location.id,
+            "conflict_resolvable": False,
+        })
+    deleted_relation_message = " associated with this asset has been deleted. "
+    if asset_cp.model is None:
+        conflicts.append({
+            "conflict_message":
+                "The model" + deleted_relation_message + nonresolvable_message,
+            "conflicting_asset": None,
+            "conflict_resolvable": False,
+        })
+    if asset_cp.rack is None:
+        conflicts.append({
+            "conflict_message":
+                "The rack" + deleted_relation_message + nonresolvable_message,
+            "conflicting_asset": None,
+            "conflict_resolvable": False,
+        })
+    if asset_cp.related_decommissioned_asset:
+        conflicts.append({
+            "conflict_message":
+                "This asset has been decommissioned in a live change. " +
+                nonresolvable_message,
+            "conflicting_asset": None,
+            "conflict_resolvable": False,
+        })
+    if len(conflicts) == 0:
+        if asset_cp.is_conflict:
+            conflicts.append({
+                "conflict_message":
+                    "Live changes have been made to this asset since your " +
+                    "latest change planner modification. This conflict " +
+                    "needs to be resolved. Please select which version " +
+                    "you would like to keep.",
+                "conflicting_asset": None,
+                "conflict_resolvable": True,
+            })
```

Comment:
based on my comments on the slack thread, `if asset_cp.is_conflict && asset_cp.related_asset is None`: to check if the asset has been deleted on msater. and then mark this as a unresolvable conflict? conflict message `The asset being modified on this change plan has been deleted live. + non resolvable message`